docs/pages/_app.js

```diff
@@ -1,3 +1,11 @@
+/* eslint-disable import/first */
+import { LicenseInfo } from '@mui/x-data-grid-pro';
+
+// Remove the license warning from demonstration purposes
+LicenseInfo.setLicenseKey(
+  '0f94d8b65161817ca5d7f7af8ac2f042T1JERVI6TVVJLVN0b3J5Ym9vayxFWFBJUlk9MTY1NDg1ODc1MzU1MCxLRVlWRVJTSU9OPTE=',
+);
```

Comment:
I have asked the x team about this, they confirm that this pattern is okay. However, @eps1lon suggest to take a look at how Remix tackle the license. I agree that it can be improved and will open an issue in the x repo.

Reply:
We could get the license key from an env var: https://docs.netlify.com/configure-builds/environment-variables/

Reply:
For context:

- Remix uses a private npm repository
- We use the license key as a reminder for developers to know they are not licensed, it's not a hard constraint. The hard constraint is the international IP laws.

Regarding the advantages of a public repository, I can think of two:

- We get social proofs, potential customers can see if we are a leading solution in the market or not https://npm-stat.com/charts.html?package=%40material-ui%2Fx-grid
- The trial onboarding is smooth, there are no free trial forms to fill, no need to leave your personal information.

Regarding the downsides:

- We can't enforce that everybody is paying, it's going to be abused. In some way, it can be great, these developers abusing the license for a side project might be more likely to get his employee pay
- We don't collect lead information, people a sales team could call back, to gather feedback on how well it went. Now, Telerik still has a public npm repository and asks for a free trial to go to a form to be licensed, we could always do the same.

Reply:
> Just to clarify for me: We actually require leaking license keys to client bundle? How would we prove to a buyer that they shared the key illegally when it's just as likely that someone else just read the license key from their website?

@eps1lon In the current approach: we require leaking license keys to client bundles, correct. The license key is a reminder to purchase a license. For instance, it's not because a project has a valid license key that it's licensed. 

Regarding the prior-arts, there were three influences:

- https://fullcalendar.io/license
- https://ag-grid.com/react-data-grid/licensing/#setting-the-license-key

More recently, I saw a new approach in https://www.telerik.com/kendo-react-ui/my-license/, we could also investigate it.

Reply:
> Regarding the prior-arts, there were three influences:

And why did you choose these approaches? Did you just choose them because somebody else is doing it? How can this possibly be a meassure of a "good" approach.

Also: I moved this discussion explicitly to internal channels since I realized how irresponsible it is to have this security related discussion in public.

Reply:
The approach in https://www.telerik.com/kendo-react-ui/my-license/ is interesting:

- `@progress/kendo-licensing` read the public key of the license during the build phase
- once the license is read and validated with their public key, `@progress/kendo-licensing` updates the [dist files](https://unpkg.com/@progress/kendo-licensing@1.2.0/dist/index.js) at the `data: '  {}  '` placeholder.
- once this is done, the library is built without containing the license key, it stays private on the server.

Regarding the cons:

- Looking at the source, it might be a bit tricky to do correctly
- It forces a build step, it's very common to have one, but not always

Reply:
Regarding the security aspect, the license key **can** be made public.

Reply:
I have opened https://github.com/mui-org/material-ui-x/issues/2565 for it.